Sources/CoreAardvark/PrivateCategories/NSFileHandle+ARKAdditions.m

```diff
@@ -42,10 +42,15 @@ - (void)ARK_writeDataBlock:(NSData *)dataBlock;
     uint8_t dataLengthBytes[ARKBlockLengthBytes] = { };
     ARKWriteBigEndianBlockLength(dataLengthBytes, 0, dataBlockLength);
     NSData *dataLengthData = [NSData dataWithBytes:dataLengthBytes length:ARKBlockLengthBytes];
-    
+
+    NSMutableData *combinedData = [dataLengthData mutableCopy];
+    [combinedData appendData:dataBlock];
+
+    __weak NSData *weakCombinedData = combinedData;
     @try {
-        [self writeData:dataLengthData];
-        [self writeData:dataBlock];
+        @autoreleasepool {
+            [self writeData:weakCombinedData];
+        }
```

Comment:
This syntax is equivalent to:

```objc
NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
[self writeData:weakCombinedData];
[pool release];
```

Given that `writeData` can throw, it's easy to see that in that case we'd leak the autorelease pool (and with that, every autoreleased object).

There's really no safe way to catch Obj-C exceptions under ARC. Even doing what I suggested, wrapping `-writeData:` in a file that we compile without ARC would leak objects, because the code inside Foundation's method doesn't expect that the exception is throws will be caught.
I think we'd have to be OK with leaking memory in this scenario tbh.